src/sinks/aws_kinesis/streams/integration_tests.rs

```diff
@@ -23,62 +26,71 @@ fn kinesis_address() -> String {
-//        region: RegionOrEndpoint::with_both("localstack", kinesis_address().as_str()),
-//        encoding: TextSerializerConfig::default().into(),
-//        compression: Compression::None,
-//        request: Default::default(),
-//        tls: Default::default(),
-//        auth: Default::default(),
-//        acknowledgements: Default::default(),
-//    };
-//
-//    let partition_key: String = "partition_key".to_string();
-//
-//    let config = KinesisStreamsSinkConfig {
-//        partition_key_field: Some(partition_key.clone()),
-//        batch,
-//        base,
-//    };
-//
-//    let cx = SinkContext::default();
-//
-//    let sink = config.build(cx).await.unwrap().0;
-//
-//    let timestamp = chrono::Utc::now().timestamp_millis();
-//
-//    let (mut input_lines, events) = random_lines_with_stream(100, 11, None);
-//
-//    run_and_assert_sink_compliance(sink, events, &AWS_SINK_TAGS).await;
-//
-//    sleep(Duration::from_secs(1)).await;
-//
-//    let records = fetch_records(stream, timestamp).await.unwrap();
-//
-//    let mut output_lines = records
-//        .into_iter()
-//        .map(|e| {
-//            assert_eq!(Some(partition_key.as_str()), e.partition_key());
-//            e
-//        })
-//        .map(|e| String::from_utf8(e.data.unwrap().into_inner()).unwrap())
-//        .collect::<Vec<_>>();
-//
-//    input_lines.sort();
-//    output_lines.sort();
-//    assert_eq!(output_lines, input_lines)
-//}
+#[tokio::test]
+async fn kinesis_put_records_with_partition_key() {
+    let stream = gen_stream();
+
+    ensure_stream(stream.clone()).await;
+
+    let mut batch = BatchConfig::default();
+    batch.max_events = Some(2);
+
+    let partition_value = "a_value";
+
+    let base = KinesisSinkBaseConfig {
+        stream_name: stream.clone(),
+        region: RegionOrEndpoint::with_both("localstack", kinesis_address().as_str()),
+        encoding: TextSerializerConfig::default().into(),
+        compression: Compression::None,
+        request: Default::default(),
+        tls: Default::default(),
+        auth: Default::default(),
+        acknowledgements: Default::default(),
+        request_retry_partial: Default::default(),
+    };
+
+    let partition_key = ConfigValuePath::try_from("partition_key".to_string()).unwrap();
+
+    let config = KinesisStreamsSinkConfig {
+        partition_key_field: Some(partition_key.clone()),
+        batch,
+        base,
+    };
+
+    let cx = SinkContext::default();
+
+    let sink = config.build(cx).await.unwrap().0;
+
+    let timestamp = chrono::Utc::now().timestamp_millis();
+
+    let (mut input_lines, events) = random_lines_with_stream(100, 11, None);
+
+    let events = events.map(move |mut events| {
+        events.iter_logs_mut().for_each(move |log| {
+            log.insert("partition_key", partition_value);
+        });
+        events
+    });
+
+    run_and_assert_sink_compliance(sink, events, &AWS_SINK_TAGS).await;
+
+    sleep(Duration::from_secs(1)).await;
```

Comment:
I understand this was part of the original test but it is a code smell (slower tests, flakiness etc). Can we add a comment here explaining why this is needed?

Reply:
I believe this test case was copied by the author from the one below it.

In searching the code for `run_and_assert_sink_compliance` it looks like most of the aws integration tests using this function, have a sleep like this.
Luckily I was able to find one comment so I will add that here.